src/multi-select/multi-select.component.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { PropTypes } from 'react';
+import { Map } from 'immutable';
+import ImmutablePropTypes from 'react-immutable-proptypes';
+
+import MultiSelectItem from './multi-select-item/multi-select-item.component';
+import './multi-select.component.scss';
+
+export default class MultiSelect extends React.PureComponent {
+
+  static propTypes = {
+    items: PropTypes.arrayOf(PropTypes.shape({
+      id: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.number,
+      ]).isRequired,
+      text: PropTypes.string.isRequired,
+    })).isRequired,
+    checkedItems: ImmutablePropTypes.map,
+    onChange: PropTypes.func,
+  };
+
+  static defaultProps = {
+    checkedItems: Map(),
+    onChange: () => {},
+  };
+
+  isChecked = (id, checkedItems) => checkedItems.get(id) === true;
+
+  render() {
+    const { items, checkedItems, onChange } = this.props;
+    return (
+      <div className="oc-multi-select">
+        {items.map((item) => {
+          const { id, text, ...otherProps } = item;
+          const childItem = { id, text };
```

Comment:
Object literal might cause the problem with PureComponent in multi-select-item since it creates a new object and hence the shallow comparison evaluates to true and re-render happens. The item should be passed directly to multi-select-item component  and destructuring done there.